autonomi client simplification

joshuef opened the pull request.

oetyng commented on sn_node/src/node/flow_ctrl/periodic_checks.rs:
```diff
@@ -143,26 +143,19 @@ impl FlowCtrl {
 
         // check if we can join the dst section
-        if let Some(RelocationState::ReadyToJoinNewSection(proof)) = &context.relocation_state {
+        if let Some(proof) = context.relocation_state.proof() {
```
I think it was neat and good to remove the Option as you did. We could be more consistent if we do just as on line 121, or indeed as it was before, and call:

```
if let Some(RelocationState::ReadyToJoinNewSection(proof)) = &context.relocation_state {
```

Or what do you think?
Then we don't need the extra

```
impl RelocationState {
    pub fn proof(&self) -
```

stuff either.

oetyng replied:
I see the `fn proof() `is reused further down, but it makes it clearer I think to make it explicit that it is when `RelocationState::ReadyToJoinNewSection(proof)` that we call `node.finalise_relocation`.

oetyng commented on sn_node/src/node/messaging/node_msgs.rs:
```diff
@@ -205,6 +204,11 @@ impl MyNode {
                             target_sap.prefix(),
                         );
 
+                        // If we were relocating finaise it
```
Small typo: `finaise` => `finalise`

joshuef committed:
chore(aut): refactor away client traits

seems like the Wallet is the interface we might need traits over...

even then if we're not exposing that to the world we can just use feat-gate
imports here over the APIs needed, I think

grumbach reviewed (approved):
Excellent! LGTM
